packages/sdk/src/types.ts

```diff
@@ -1170,7 +1178,18 @@ export interface AccountInfoDelegator extends AccountInfoCommon {
     accountDelegation: AccountDelegationDetails;
 }
 
-export type AccountInfo = AccountInfoSimple | AccountInfoBaker | AccountInfoDelegator;
+export interface AccountInfoUnknown extends AccountInfoCommon {
+    type: AccountInfoType.Unknown;
+    /**
+     * This field is named `accountBaker` to align with the JSON representation produced by the
+     * corresponding rust SDK.
+     * `null` represents a variant of staking info for the account which is unknown to the SDK, otherwise
+     * this will always be `undefined`, as otherwise this is represented by {@linkcode AccountInfoBaker}.
+     */
+    accountBaker: Upward<never>;
```

Comment:
Maybe I just don't understand fully when this field can/should be `undefined`. I understand that if someone wants to record real `accountBaker` data the type `AccountInfoBaker` should be used and not the `AccountInfoUnkown` even it has this name field.

Reply:
sorry forgot to update the comment, you're right.. good catch!